### ISSUE TITLE
Twisted from Scratch: Evolution of Finger (haifux talk/howto)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4839]
|<img alt="zetet's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| zetet reported|
|-|-|
|Trac ID|trac#319|
|Type|enhancement|
|Created|2003-10-07 08:30:25Z|

<details><summary>Searchable metadata</summary>

```
trac-id__319 319
type__enhancement enhancement
reporter__zetet zetet
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__ 
keywords__ 
time__1065515425000000 1065515425000000
changetime__1469033265775542 1469033265775542
version__ 
owner__ 
cc__itamarst cc__zetet
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="zetet's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|zetet commented|
|-|-|

```
#!html
<pre>
I took the haifux talk from sandbox/jml/haifux and made the examples
work, also updated the APIs. 

http://zetet.com/twisted/finger/
http://zetet.com/twisted/finger.tar
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="zetet's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|zetet commented|
|-|-|

```
#!html
<pre>
future plans:

split into at least two parts.  Probably at the woven boundary,
so woven, PB, etc are in part two.  Part two would start around
"Use Woven"/finger20.py

show examples for users logging in to set persistent finger status
over the web.

write unit tests in order to track API changes

write docstrings for the finger module
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="zetet's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|zetet commented|
|-|-|

```
#!html
<pre>
The lore source is at 
http://www.zetet.com/twisted/finger/tour.html
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="moshez's avatar" src="https://avatars.githubusercontent.com/u/422703?s=50" width="50" height="50">](https://github.com/moshez)|@moshez commented|
|-|-|

```
#!html
<pre>
Ugh, why is this a bug?
Downgrading to feature
</pre>
```

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)|@itamarst commented|
|-|-|

```
#!html
<pre>
I'd like to commit this, but this source is somewhat of a mess. I'd like a
tarball that has only the source lore file and the listings - no temp files,
foo~ files, taps, output files, notes etc., cause I dunno what to commit and
what not to.
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="zetet's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|zetet commented|
|-|-|

```
#!html
<pre>
http://zetet.com/twisted/finger.tar
updated to clean out junk. No notes, taps, foo~ files, .pid files ...
</pre>
```

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_8"></a>|GitHub commented|
|-|-|

In changeset ba2ae0488b69d4ab85f232aa642d4120c73f49ac
```
#!CommitTicketReference repository="" revision="ba2ae0488b69d4ab85f232aa642d4120c73f49ac"
Merge pull request #319 from twisted/8592-travis-osx

Author: adiroiban
Reviewer: hawkowl
Fixes: #8592
```
